src/runtime/crt/common/crt_runtime_api.c

```diff
@@ -588,6 +591,8 @@ tvm_crt_error_t RunTimeEvaluator(tvm_function_index_t function_index, TVMValue*
       if (err != kTvmErrorNoError) {
         goto release_and_return;
       }
+      if (std::fpclassify(curr_res_seconds) == FP_ZERO) absolute_zero_times++;
+      if (absolute_zero_times >= max_repeat_num) break;
     } while (curr_res_seconds < min_repeat_seconds);
```

Comment:
It seems logical to me to keep the conditions interrupting the cycle together, unless they are doing something specific.
mb change 
```
if (absolute_zero_times >= max_repeat_num) break;
} while (curr_res_seconds < min_repeat_seconds);
```
to
`} while (curr_res_seconds < min_repeat_seconds && absolute_zero_times < max_repeat_num);`

And do it in all 3 places.

Reply:
Done